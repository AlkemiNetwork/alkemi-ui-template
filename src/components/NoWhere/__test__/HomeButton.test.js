import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import Button from "@material-ui/core/Button";

import NoWhereHomeButton from "../HomeButton";

describe("<BackHomeButton />", () => {
    let wrapper;
    let props;

    beforeEach(() => {
        props = {
            onClick: jest.fn()
        };
        wrapper = shallow(<NoWhereHomeButton {...props} />).dive();
    });

    it("renders 1 Button", () => {
        expect(wrapper.find(Button).length).toEqual(1);
    });

    it("Button is functional", () => {
        wrapper
            .find(Button)
            .at(0)
            .simulate("click");
        expect(props.onClick.mock.calls.length).toEqual(1);
    });

    it("matches snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
