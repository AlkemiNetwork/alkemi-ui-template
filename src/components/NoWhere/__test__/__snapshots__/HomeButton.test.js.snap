// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackHomeButton /> matches snapshot 1`] = `
<WithStyles(Button)
  className="NoWhereHomeButton-button-1"
  color="default"
  onClick={[MockFunction]}
  variant="raised"
>
  Bring me back home!
</WithStyles(Button)>
`;
